fastlane/lib/fastlane/actions/build_app.rb

```diff
@@ -0,0 +1,142 @@
+                                                                export_method: values[:export_method])
+
+            values[:export_options][:provisioningProfiles] = hash_to_use
+          else
+            self.show_xcode_9_warning
+          end
+        elsif Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS]
+          # Since Xcode 9 you need to explicitly provide the provisioning profile per app target
+          # If the user used sigh we can match the profiles from sigh
+          values[:export_options] ||= {}
+          if values[:export_options].kind_of?(Hash)
+            # It's not always a hash, because the user might have passed a string path to a ready plist file
+            # If that's the case, we won't set the provisioning profiles
+            # see https://github.com/fastlane/fastlane/issues/9684
+            values[:export_options][:provisioningProfiles] ||= {}
+            Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS].each do |profile_path|
+              begin
+                profile = FastlaneCore::ProvisioningProfile.parse(profile_path)
+                profile_team_id = profile["TeamIdentifier"].first
+                next if profile_team_id != values[:export_team_id] && !values[:export_team_id].nil?
+                bundle_id = profile["Entitlements"]["application-identifier"].gsub("#{profile_team_id}.", "")
+                values[:export_options][:provisioningProfiles][bundle_id] = profile["Name"]
+              rescue => ex
+                UI.error("Couldn't load profile at path: #{profile_path}")
+                UI.error(ex)
+                UI.verbose(ex.backtrace.join("\n"))
+              end
+            end
+          else
+            self.show_xcode_9_warning
+          end
+        end
+
+        gym_output_path = Gym::Manager.new.work(values)
+        if gym_output_path.nil?
+          UI.important("No output path received from gym")
+          return nil
+        end
+
+        absolute_ipa_path = File.expand_path(gym_output_path)
+        absolute_dsym_path = absolute_ipa_path.gsub(".ipa", ".app.dSYM.zip")
+
+        # This might be the mac app path, so we don't want to set it here
+        # https://github.com/fastlane/fastlane/issues/5757
+        if absolute_ipa_path.include?(".ipa")
+          Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] = absolute_ipa_path
+          ENV[SharedValues::IPA_OUTPUT_PATH.to_s] = absolute_ipa_path # for deliver
+        end
+
+        Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = absolute_dsym_path if File.exist?(absolute_dsym_path)
+        Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE] = Gym::BuildCommandGenerator.archive_path
+        ENV[SharedValues::DSYM_OUTPUT_PATH.to_s] = absolute_dsym_path if File.exist?(absolute_dsym_path)
+
+        return absolute_ipa_path
+      end
+
+      def self.description
+        "Easily build and sign your app (via gym)"
```

Comment:
What would the Android equivalent be here?

Reply:
Hmm, good question - perhaps this maps to the `gradle` action?  In which case, we should probably have `build_ios_app` and `build_android_app` (a la screenshots), and we will introduce a `build_app` action that maps to the correct behavior based on platform?

Reply:
This sounds AWESOME 👍

Reply:
I'm a little bit wary of having `build_android_app` do the same thing as `gradle`, since `gradle` does more than just build apps. What do you think, @janpio ?

Reply:
I actually didn't know that - until now I thought `gradle` was also only for building.
Is `gym` limited to absolutely only building?